### PR TITLE
Oauth start

### PR DESCRIPTION
**Pull request by AntonPuko**
just start digging. its a very first step

### CODE REVIEW

File: src/lib/PasswordHasher.js [hunk 1]
```diff
@@ -14,24 +14,37 @@
...
-	});
-};
+  static hash(
+    password: Buffer,
+    salt: Buffer,
+    callback: (error: ?Error, key: string) => void,
+  ) {
+    crypto.pbkdf2(
+      password.toString('base64'),
+      salt.toString('base64'),
```
Comment (AntonPuko) [inline src/lib/PasswordHasher.js]: pass and salt here are Buffers, but do we really need making toString() for buffers? seems in some step they transforms automatically and it works the same without toString()..hm..i don't get it yet.

Comment (jlkalberer) [inline src/lib/PasswordHasher.js]: It *works* without doing the toString but I'm guessing that there are no `flow` signatures that match passing in a `Buffer`

File: src/lib/PasswordHasher.js [hunk 2]
```diff
@@ -14,24 +14,37 @@
...
+*
 */
 
-var crypto = require('crypto');
-
-var PasswordHasher = function () {
+import crypto from 'crypto';
 
-};
+const HASH_DIGEST = 'sha1';
```
Comment (jlkalberer) [inline src/lib/PasswordHasher.js]: Make sure it's the same hash as what we were using before.

We may change this to a stronger hashing algorithm if the hardware supports it.

Comment (jlkalberer) [inline src/lib/PasswordHasher.js]: Yeah.. so make sure that `sha1` was what they were using before.  There could be code in here that expects a certain algorithm.

**Pull request merged**